core/Piranha.AspNetCore/Http/RoutingMiddleware.cs

```diff
@@ -169,10 +180,11 @@ public override async Task Invoke(HttpContext context, IApi api, IApplicationSer
                 //
                 // 4: Get the current page
                 //
-                PageBase page = null;
                 PageType pageType = null;
-
-                if (segments.Length > pos)
+                if (page != null)
+                {
```

Comment:
What is the meaning of this empty `if` block?

Reply:
To short circuit the search for a page by slug since we already set the page using the Page Id